stdlib/Dates/src/periods.jl

```diff
@@ -451,11 +451,11 @@ toms(c::Second)      = 1000 * value(c)
 toms(c::Minute)      = 60000 * value(c)
 toms(c::Hour)        = 3600000 * value(c)
 toms(c::Period)      = 86400000 * days(c)
-toms(c::CompoundPeriod) = isempty(c.periods) ? 0.0 : Float64(sum(toms, c.periods))
```

Comment:
`float` can be a bit more generic here for the same result. For example:

```
tons(c::Dates.CompoundPeriod) = isempty(c.periods) ? float(zero(eltype(c.period))) : float(sum(tons, c.periods))
```

gives a bit nicer IR:

```
julia> @code_warntype tons(p)
MethodInstance for tons(::Dates.CompoundPeriod)
  from tons(c::Dates.CompoundPeriod) @ Main REPL[4]:1
Arguments
  #self#::Core.Const(tons)
  c::Dates.CompoundPeriod
Body::Union{}
1 ─ %1  = Main.isempty::Core.Const(isempty)
│   %2  = Base.getproperty(c, :periods)::Vector{Period}
│   %3  = (%1)(%2)::Bool
└──       goto #3 if not %3
2 ─       Main.float
│         Main.zero
│         Main.eltype
│         Base.getproperty(c, :period)
│         Core.Const(:((%7)(%8)))
│         Core.Const(:((%6)(%9)))
│         Core.Const(:((%5)(%10)))
└──       Core.Const(:(return %11))
3 ┄       Main.float
│   %14 = Main.sum::Core.Const(sum)
│   %15 = Main.tons::Core.Const(tons)
│   %16 = Base.getproperty(c, :periods)::Vector{Period}
│         (%14)(%15, %16)
│         Core.Const(:((%13)(%17)))
└──       Core.Const(:(return %18))
```

Reply:
Woops, I had a typo here. This is wrong, let me check again!